doc/source/tutorial/stats/outliers.md

```diff
@@ -0,0 +1,521 @@
+
+No, not really.
+
+So at the time of writing, SciPy offers three or four ways of performing some operations related to trimming, but none are able to perform *all* operations we might reasonably expect them to. In the spirit of the Zen of Python, we suggest the new "one-- and preferably only one --obvious way to do it.": manually, in one or two lines of code, using more familiar, general-purpose functions. Whether calculating a trimmed or winsorized statistic, there will be three steps:
+
+1. Identify the outlier threshold.
+2. Trim or winsorize.
+3. Compute the statistic.
+
++++
+
+# Identifying the threshold
+
+SciPy now offers a `quantile` function with several methods useful for identifying thresholds beyond which data are to be trimmed or winsorized. The most common thresholds are those corresponding with a certain *percentage* of the data in each tail. To set the threshold at $p=10\%$ of the data in each tail, rounding the number of points to trim *down*:
+
+```{code-cell} ipython3
+p = 0.1
+stats.quantile(x, [p, 1-p], method='winsor_less')
+```
+
+Rounding the number of points to trim *up*:
+
+```{code-cell} ipython3
+stats.quantile(x, [p, 1-p], method='winsor_more')
+```
+
+Rounding the number of points to trim to the nearest integer:
+
+```{code-cell} ipython3
+stats.quantile(x, [p, 1-p], method='winsor_round')
+```
+
+Note that these methods are all subtlely different from estimating the 10th and 90th percentiles of the data. Using the default method:
+
+```{code-cell} ipython3
+stats.quantile(x, [p, 1-p])
+```
+
+The Harrel-Davis method:
+
+```{code-cell} ipython3
+stats.quantile(x, [p, 1-p], method='harrell-davis')
+```
+
+Note that `quantile` is fully vectorized. Suppose we have two rows of data, and we wish to set different thresholds for each row.
+
+```{code-cell} ipython3
+x2d = np.stack([x, x])
+p2d = [[p  , 1 - p  ],  # thresholds for zeroth row
+       [2*p, 1 - 2*p]]  # thresholds for next row
+stats.quantile(x2d, p2d, method='winsor_less', axis=-1)
+```
+
+## Missing Data
+
++++
+
+If there are values to be ignored, use an [MArray](https://mdhaber.github.io/marray/tutorial.html) to mask them out. 
+
+```{code-cell} ipython3
+from marray import numpy as mxp
+mask = ((x2d % 2) == 0)  # ignore odd elements
+y = mxp.asarray(x2d, mask=mask)
+print(y)
+```
+
+`quantile` and many other `scipy.stats` functions natively support `MArray`s.
+
+```{code-cell} ipython3
+stats.quantile(y, mxp.asarray(p2d), method='winsor_less', axis=-1)
+```
+
+If NaNs are used to represent missing values, the `nan_policy='omit'` option is also available
+
+```{code-cell} ipython3
+z = x2d.copy()
+z[mask] = np.nan
+stats.quantile(z, p2d, method='winsor_less', nan_policy='omit', axis=-1)
+```
+
+# Trimming and Winsorizing
+
++++
+
+## Winsorizing is `clip`ing
+
++++
+
+Once a numerical threshold has been identified, winsorizing is identical to a routine so fundamental that it is part of the array API standard: `clip`.
```

Comment:
Should probably link to the array API standard. Probably need to explain that `SCIPY_ARRAY_API=1` needs to be set to do the things in this tutorial.